modules/crm_core_default_matching_engine/src/Plugin/crm_core_match/engine/DefaultMatchingEngine.php

```diff
@@ -27,7 +29,6 @@
  *   priority = 0,
  *   settings = {
```

Comment:
I don't really know what this is doing, so I think this can go completely? The standard way to have default configuration for configurable plugins is to implement defaultConfiguration(), although it needs a bit of boilerplate code to actually ensure that's used, see BlockBase.